iceoryx_platform/generic/source/stdlib.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 by Mathias Kraus <elboberido@m-hias.de>. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_platform/stdlib.hpp"
+
+#ifndef IOX_PLATFORM_OVERRIDE_STDLIB_ALL
+
+#include <cerrno>
+#include <cstring>
+#include <mutex>
+
+namespace
+{
+std::mutex& env_mutex()
+{
+    static std::mutex mtx;
+    return mtx;
+}
+} // namespace
+
+#ifndef IOX_PLATFORM_OVERRIDE_STDLIB_GETENV_S
+
+int iox_getenv_s(size_t* actual_size_with_null, char* buffer, size_t buffer_capacity, const char* name)
+{
+    std::lock_guard<std::mutex> lock(env_mutex());
+
+    if (name == nullptr)
```

Comment:
Shouldn't we also make sure that `actual_size_with_null` is not a `nullptr`? If this were `nullptr` would the user be able to use this cleanly since they do not know how large `buffer` is.

I know it is null terminated but could it be possible that `\0` is also actually a value?!

Reply:
If I understood the documentation correctly, then that parameter can be a nullptr. Please also check https://en.cppreference.com/w/c/program/getenv

AFAIK only strings are allowed as env variables therefore it should not be an issue.